include/git2/hook.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#ifndef INCLUDE_git_hook_h__
+#define INCLUDE_git_hook_h__
+
+#include "common.h"
+#include "types.h"
+
+/**
+ * @file git2/hook.h
+ * @brief Git hook management routines
+ * @defgroup git_hook Git hook management routines
+ * @ingroup Git
+ * @{
+ */
+GIT_BEGIN_DECL
+
+/**
+ * A helper to get the path of the repository's hooks.
+ *
+ * This will obey core.hooksPath.
+ *
+ * @param out_dir The absolute path to the hooks location.
+ * @param repo The repository to get the hooks location from.
+ * @return 0 on success, or an error code.
+ */
+GIT_EXTERN(int) git_hook_dir(char **out_dir, git_repository *repo);
```

Comment:
You should make clear that `out_dir` is caller-owned (at least I guess it is due to it not being `const`). Seeing that we might have custom allocators, we probably shouldn't even pass back memory ownership, as we do not expose `git__free`. Maybe use a `git_buf` instead, as there we have `git_buf_free`?